include/wlr/types/wlr_xdg_shell.h

```diff
@@ -163,7 +163,8 @@ struct wlr_xdg_surface {
 	struct wlr_box next_geometry;
 	struct wlr_box geometry;
 
-	struct wl_listener surface_destroy_listener;
+	struct wl_listener surface_destroy;
+	struct wl_listener surface_commit;
```

Comment:
Why do we have two surface commit handlers now?

Reply:
`xdg-shell` wants to do stuff before the role has been assigned (mainly check that there's no non-null commit before configure has been sent). So, we need another listener to do that.